app/views/handlers/list_action.py

```diff
@@ -55,6 +55,10 @@ def get_next_location_url(self):
         return self.parent_location.url()
 
     def handle_post(self):
+        self.questionnaire_store_updater.update_same_name_items(
+            self.parent_block["for_list"],
+            self.parent_block.get("same_name_answer_ids", []),
```

Comment:
The default of `[]` isn't required given `update_same_name_items` exits early if `same_name_answer_ids` is falsey and doesn't try iterate by default. The type hint should be updated 
```py
same_name_answer_ids: Optional[List[str]]
```
Applies to all occurrences.